packages/acceptance-tests/pkg-tests-specs/sources/commands/add.test.js

```diff
@@ -116,5 +117,50 @@ describe(`Commands`, () => {
         });
       }),
     );
+
+    test(`it should clean the cache when cache lives inside the project`, makeTemporaryEnv({
+      dependencies: {
+        [`no-deps`]: `1.0.0`,
+      },
+    }, async ({path, run, source}) => {
+      let code;
+      let stdout;
+      let stderr;
+
+      await run(`install`);
+
+      expect(await readdir(`${path}/.yarn/cache`)).toMatchSnapshot();
+
+      ({ code, stdout, stderr } = await run(`add`, `no-deps@2.0.0`));
+
+      await expect({code, stdout, stderr}).toMatchSnapshot();
+
+      expect(await readdir(`${path}/.yarn/cache`)).toMatchSnapshot();
+    }));
+
+    test(`it should not clean the cache when cache lives outside the project`, makeTemporaryEnv({
+      dependencies: {
+        [`no-deps`]: `1.0.0`,
+      },
+    }, async ({path, run, source}) => {
+      const sharedCachePath = await createTemporaryFolder();
+      const env = {
+        YARN_CACHE_FOLDER: sharedCachePath
+      };
+
+      let code;
+      let stdout;
+      let stderr;
+
+      await run(`install`, {env});
+
+      expect(await readdir(sharedCachePath)).toMatchSnapshot();
```

Comment:
I'd prefer to avoid using snapshots if possible - can you replace the `readdir` snapshots by a number of files check?